packages/common/src/utils/filters.ts

```diff
@@ -102,7 +104,7 @@ export const getFilterGroupItemsPropertyName = (
 };
 
 export const getFilterTypeFromItem = (item: FilterableField): FilterType => {
-    const { type } = item;
+    const type = isCustomSqlDimension(item) ? item.dimensionType : item.type;
```

Comment:
at this point we should probably create a `getDimensionType` method that takes a Dimension or CustomSqlDimension

Reply:
Was thinking the same, doing it after this PR